### PR TITLE
Custom check suport

### PR DESCRIPTION
**Pull request by vengestone-dragoon**

### CODE REVIEW

File: src/main/java/com/osiris/autoplug/client/tasks/updater/search/CustomCheckURL.java [hunk 1]
```diff
@@ -39,7 +39,11 @@ public SearchResult doCustomCheck(String url, String currentVersion) {
                 release = Json.getAsJsonArray(url)
                         .get(0).getAsJsonObject();
             } catch (Exception e) {
-                throw e;
+                try {
+                    release = Json.getAsJsonObject(url);
+                } catch (Exeption ex){
+                    throw ex;
+                }
```
Comment (Osiris-Team) [inline src/main/java/com/osiris/autoplug/client/tasks/updater/search/CustomCheckURL.java:46]: you should be able to check the jsonElement to see if its a object or array further above before accessing it.
What about the key traversal thing btw?

**Review by Osiris-Team — CHANGES_REQUESTED** [review]

File: src/main/java/com/osiris/autoplug/client/tasks/updater/search/CustomCheckURL.java (hunk 1 above)
Comment (vengestone-dragoon) [reply src/main/java/com/osiris/autoplug/client/tasks/updater/search/CustomCheckURL.java:51]: it now checks if its an object or array
i have looked but am failing to find a way to traverse the keys that i understand well enough to try to implement

Comment (vengestone-dragoon) [reply src/main/java/com/osiris/autoplug/client/tasks/updater/search/CustomCheckURL.java]: done

**Comment by Osiris-Team** [comment]
I'll merge this and enhance the traversal function a bit, its doesn't need to be that complicated ;)

**Pull request merged**